src/LiveSplit.WorldRecord/UI/Components/WorldRecordComponent.cs

```diff
@@ -247,19 +259,35 @@ private void ShowWorldRecord(LayoutMode mode)
         }
     }
 
-    private bool IsPBTimeLower(TimeSpan? pbTime, TimeSpan? recordTime, bool showMillis)
+    private bool CheckPrecisionMillis(TimeSpan? recordTime)
+    {
+        if(Settings.WRPrecision == WorldRecordPrecisionType.FromLeaderboard)
+        {
+            if(!recordTime.HasValue)
+            {
+                // Fallback to Milliseconds
+                return true;
+            }
+
+            return recordTime.Value.Milliseconds != 0;
```

Comment:
This will cause a bug if the world record is `1:23.000` and the PB time is `1:23.999`; it'll show that you have the record because `IsPBTimeLower` will only check the seconds.

Reply:
You described the behavior as "Follow the leaderboard setting", but I don't think `game.Ruleset.ShowMilliseconds` is being used anymore?

Reply:
It's a tough call as to which behavior would be better - a 1/1000 chance of a bug, versus getting the correct behavior for games that allow submitting milliseconds but have categories that only allow seconds.

Reply:
The current behavior in this PR is probably fine.

Reply:
Yeah, the gamerule has to be set as soon a a single category uses milliseconds. We could create a function in the Sharp API repo that checks if any of the runs on the leaderboard of interest has non-zero milliseconds to further make running into this bug much more unlikely than 1/1000, but I think this is not needed.